src/algorithms/optimization/fixed_point_differentiation.jl

```diff
@@ -1,99 +1,167 @@
+    - `:gmres` : GMRES iterative linear solver, see the [KrylovKit docs](https://jutho.github.io/KrylovKit.jl/stable/man/algorithms/#KrylovKit.GMRES) for details
+    - `:bicgstab` : BiCGStab iterative linear solver, see the [KrylovKit docs](https://jutho.github.io/KrylovKit.jl/stable/man/algorithms/#KrylovKit.BiCGStab) for details
 """
 struct LinSolver{F} <: GradMode{F}
-    solver::KrylovKit.LinearSolver
-end
-function LinSolver(;
-    solver=KrylovKit.BiCGStab(; maxiter=Defaults.fpgrad_maxiter, tol=Defaults.fpgrad_tol),
-    iterscheme=Defaults.iterscheme,
-)
-    return LinSolver{iterscheme}(solver)
+    solver_alg::KrylovKit.LinearSolver
 end
+LinSolver(; kwargs...) = GradMode(; alg=:linsolver, kwargs...)
+
+GRADIENT_MODE_SYMBOLS[:linsolver] = LinSolver
 
 """
-    struct EigSolver(; solver=KrylovKit.Arnoldi(), iterscheme=Defaults.iterscheme) <: GradMode{iterscheme}
+    struct EigSolver <: GradMode{iterscheme}
 
 Gradient mode wrapper around `KrylovKit.KrylovAlgorithm` for solving the gradient linear
 problem as an eigenvalue problem.
 
-With `iterscheme` the style of CTMRG iteration which is being differentiated can be chosen.
-If set to `:fixed`, the differentiated CTMRG iteration is assumed to have a pre-computed
-SVD of the environments with a fixed set of gauges. Alternatively, if set to `:diffgauge`,
-the differentiated iteration consists of a CTMRG iteration and a subsequent gauge fixing step,
-such that `gauge_fix` will also be differentiated everytime a CTMRG derivative is computed.
+## Keyword arguments
+
+* `tol::Real=$(Defaults.gradient_tol)` : Convergence tolerance of the linear solver.
+* `maxiter::Int=$(Defaults.gradient_maxiter)` : Maximal number of solver iterations.
+* `verbosity::Int=$(Defaults.gradient_verbosity)` : Output information verbosity of the linear solver.
+* `iterscheme::Symbol=:$(Defaults.gradient_iterscheme)` : Style of CTMRG iteration which is being differentiated, which can be:
+    - `:fixed` : the differentiated CTMRG iteration uses a pre-computed SVD with a fixed set of gauges
+    - `:diffgauge` : the differentiated iteration consists of a CTMRG iteration and a subsequent gauge-fixing step such that the gauge-fixing procedure is differentiated as well
+* `solver_alg::Union{KrylovKit.KrylovAlgorithm,NamedTuple}=(; alg=:$(Defaults.gradient_eigsolver)` : Eigen solver algorithm which, if supplied directly as a `KrylovKit.KrylovAlgorithm` overrides the above specified `tol`, `maxiter` and `verbosity`. Alternatively, it can be supplied via a `NamedTuple` where `alg` can be one of the following:
+    - `:arnoldi` : Arnoldi Krylov algorithm, see the [KrylovKit docs](https://jutho.github.io/KrylovKit.jl/stable/man/algorithms/#KrylovKit.Arnoldi) for details
```

Comment:
More as a reminder for the future, we should enable external references when we update the documentation.